crypto/provider_conf.c

```diff
@@ -80,6 +81,25 @@ static int provider_conf_params(OSSL_PROVIDER *prov,
 
         OSSL_TRACE1(CONF, "Provider params: start section %s\n", value);
 
+        /*
+         * Check to see if the provided section value has already
+         * been visited.  If it has, then we have a recursive lookup
+         * in the configuration which isn't valid.  As such we should error
+         * out
+         */
+        for (i = 0; i < sk_OPENSSL_CSTRING_num(visited); i++) {
+            if (sk_OPENSSL_CSTRING_value(visited, i) == value) {
+                ERR_raise(ERR_LIB_CRYPTO, ERR_R_INTERNAL_ERROR);
```

Comment:
Should this error be something more descriptive? Also ERR_R_INTERNAL_ERROR does not seem to be right to me.

Reply:
It certainly could be, but I'm not sure what currently defined error would be more appropriate.  Perhaps this error code used in ERR_raise_data with a descriptive message?